flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/utils/FlinkUtils.java

```diff
@@ -147,12 +148,7 @@ public static boolean isHaMetadataAvailable(
                         .list()
                         .getItems();
 
-        return configMaps.stream()
-                .anyMatch(
-                        map ->
-                                !map.isMarkedForDeletion()
-                                        && map.getData() != null
-                                        && !map.getData().isEmpty());
+        return configMaps.stream().anyMatch(Predicate.not(ConfigMap::isMarkedForDeletion));
```

Comment:
In case somebody wonders, this change was included in 45a280aa2d6824ded3a03b503dc33208f159f5d9.